students/cfarris/lessons/lesson09/assignment/charges_calc_2.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env Python3
+
+'''
+Lesson9 part2--edit file to use a decorator to toggle debugging on/off
+from command line
+Note: This script could use some linting. However, in the interest of time,
+I am turning in as is. I would also like to add tests later on...during break
+I apologize for the code smells!
+'''
+
+import functools
+import argparse
+import json
+import datetime
+import math
+import logging
+
+
+
+log_format = "%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s %(message)s"
+
+formatter = logging.Formatter(log_format)  #create a formatter
+
+logger = logging.getLogger()  #calls the logger object
+
+
+def parse_cmd_arguments():
+    '''
+    This function accepts parameters from user input and stores arguments until called
+    #-d should turn debugging on or off. -d True turns dbug on, -d off turns debug off.
+    #debug will be set to off
+    PARAMS: input and output are required by user, debug level is optional
+    RETURN: arguments provided by user
+    '''
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument('-o', '--output', help='ouput JSON file', required=True)
+    parser.add_argument('-d', '--debug', help='report logging debug', required=False)
+    logging.debug(parser.parse_args())
+    return parser.parse_args()
+
+
+
+def set_logger(fn):
+    '''
+    set_logger is used a a decorator to turn logging on and off. 
+    Logger is decorated and the debug argument is added from command line.
+    '''
+    def _decorator(debug, *args, **kwargs):
+        if debug == 'False':
+            result = fn(debug, *args)
+            return result
+        else:
+            file_handler = logging.FileHandler('charges_calc.log')
+            file_handler.setFormatter(formatter)
+            logger.addHandler(file_handler)
+            logger.setLevel("INFO")
+            result = fn(debug, *args)
+            return result
+    return _decorator
```

Comment:
Shouldn't you be turning logging off after the function is run? Otherwise you are changing logging globally, not just for the functions decorated. 

Using the fact that debug is a global is definitely the easiest way for the toggle, good find.